common/persistence/sql/sqlExecutionManager.go

```diff
@@ -984,17 +997,47 @@ func (m *sqlExecutionManager) CompleteReplicationTask(
 	request *p.CompleteReplicationTaskRequest,
 ) error {
 
-	if _, err := m.db.DeleteFromReplicationTasks(&sqldb.ReplicationTasksFilter{
-		ShardID: m.shardID,
-		TaskID:  &request.TaskID,
-	}); err != nil {
+	if _, err := m.db.DeleteFromReplicationTasks(m.shardID, int(request.TaskID)); err != nil {
 		return &workflow.InternalServiceError{
 			Message: fmt.Sprintf("CompleteReplicationTask operation failed. Error: %v", err),
 		}
 	}
 	return nil
 }
 
+func (m *sqlExecutionManager) GetReplicationTasksFromDLQ(
+	request *p.GetReplicationTasksFromDLQRequest,
+) (*p.GetReplicationTasksFromDLQResponse, error) {
+
+	readLevel, maxReadLevelInclusive, err := getReadLevels(&request.GetReplicationTasksRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	filter := sqldb.ReplicationTasksFilter{
+		ShardID:   m.shardID,
+		MinTaskID: readLevel,
+		MaxTaskID: maxReadLevelInclusive,
+		PageSize:  request.BatchSize,
+	}
+	rows, err := m.db.SelectFromReplicationTasksDLQ(&sqldb.ReplicationTasksDLQFilter{
+		ReplicationTasksFilter: filter,
+		SourceClusterName:      request.SourceClusterName,
+	})
+	if err != nil {
+		if err != sql.ErrNoRows {
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("GetReplicationTasks operation failed. Select failed: %v", err),
+			}
+		}
+	}
+	if len(rows) == 0 {
+		return &p.GetReplicationTasksResponse{}, nil
+	}
```

Comment:
i would rather make this a switch case, like below, and let populateGetReplicationTasksResponse deal with empty rows
```go
switch err {
case nil:
    return m.populateGetReplicationTasksResponse(rows, request.MaxReadLevel)
case sql.ErrNoRows:
    return &p.GetReplicationTasksResponse{}, nil
default:
    return err
}
```

Reply:
done